rs/tests/src/os_qualification_utils/steps/mod.rs

```diff
@@ -0,0 +1,59 @@
+use ic_system_test_driver::driver::test_env::TestEnv;
+use slog::{error, info};
+use tokio::runtime::Handle;
+
+pub mod ensure_elected_version;
+pub mod retire_elected_version;
+pub mod update_subnet_type;
+pub mod workload;
+pub mod xnet;
+
+const DEFAULT_TOTAL_RUNS: usize = 1;
+
+pub trait Step: Sync + Send {
+    fn execute(&self, env: TestEnv, rt: Handle) -> anyhow::Result<()>;
+
+    // Specify the total number of runs for a step
+    //
+    // Useful for flaky steps like workload testing or
+    // xnet testing. In general, the whole test will be
```

Comment:
I don't think XNet testing is flaky. I had an [MR to make this explicit](https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/17551) at some point but it turns out I forgot to merge it.